### PR TITLE
[WIP] Adds in process component to refactor

### PR DESCRIPTION
**Pull request by jkotalik**
Few big issues still remain before I can try this out. Will discuss appropriately.

### CODE REVIEW

**Comment by pan-wang** [comment]
could you please remove the files under bin folder?

File: src/AspNetCore/Src/applicationinfo.cpp [hunk 1]
```diff
@@ -127,7 +127,7 @@ APPLICATION_INFO::EnsureApplicationCreated()
     // load assembly and create the application
     if (m_pConfiguration->QueryHostingModel() == APP_HOSTING_MODEL::HOSTING_IN_PROCESS)
     {
-        hr = LoadAssemblyFromLocalBin();
+        hr = LoadAssemblyFromInetsrv();
```
Comment (jkotalik) [inline src/AspNetCore/Src/applicationinfo.cpp:137]: Let's have the in process scenario load from inetsrv (or IISExpress) until we have the logic to look in the runtimestore/bin.

File: src/RequestHandler/inprocessapplication.cpp [hunk 2]
```diff
@@ -1,21 +1,653 @@
...
+    _In_ IHttpContext* pHttpContext
+)
+{
+    if (m_RequestHandler != NULL)
+    {
+        return m_RequestHandler(pHttpContext, m_RequstHandlerContext);
+    }
+
+    ////
+    //// return error as the application did not register callback
```
Comment (jkotalik) [inline src/RequestHandler/inprocessapplication.cpp:141]: We need to centralize all FREB/log message to somewhere besides the forwarding handler.

File: src/RequestHandler/inprocessstoredcontext.cxx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        return E_FAIL;
+    }
+
+    if (ppInProcessStoredContext == NULL)
+    {
+        return E_FAIL;
+    }
+
+    // TODO how do get the module id?
+    *ppInProcessStoredContext = (IN_PROCESS_STORED_CONTEXT*)pHttpContext->GetModuleContextContainer()->GetModuleContext((HTTP_MODULE_ID) 100);
```
Comment (jkotalik) [inline src/RequestHandler/inprocessstoredcontext.cxx:86]: Will there be any notion of a module in the request handler?

**Review by jkotalik — COMMENTED** [review]
Few questions.

File: src/AspNetCore/Src/applicationinfo.cpp [hunk 4]
```diff
@@ -209,7 +210,9 @@ APPLICATION_INFO::LoadAssemblyFromInetsrv()
                 fDone = TRUE;
             }
         }
-        hr = struFileName.Append(L"\\aspnetcorerh.dll");
```
Comment (pan-wang) [inline src/AspNetCore/Src/applicationinfo.cpp]: I already fixed this loading part

**Review by pan-wang — APPROVED** [review]
you need to rebased the change. I already fixed the build and loadassembly part. Other part of you change looks good.

**Pull request merged**